singer_sdk/connectors/sql.py

```diff
@@ -766,6 +808,8 @@ def prepare_table(
                 as_temp_table=as_temp_table,
             )
             return
+        if self.config["load_method"] == TargetLoadMethods.OVERWRITE:
+            self.truncate_table(full_table_name=full_table_name)
```

Comment:
My main concern here would be what would happen if loading fails for some unrelated reason, then the data in the table is lost, right?

Reply:
Yeah thats a good point. I was originally thinking that since the user is choosing to overwrite completely, thats the risk theyre taking. If you overwrite and your input is empty then you get an empty table.

Although we could be nicer than that and use different workflow that renames the table instead of truncating. Then if it fails we can rollback and rename the original table back before exiting and if it succeeds we'd drop the original table completely once the sync is successful. Is that what you were thinking?

Reply:
In a future iteration we can implement this more forgiving workflow to:

1. rename existing table as a backup
2. recreate new table
3. if success then delete renamed backup table
4. if fail then rollback by deleting new table and renaming the backup table back to the original name

This is my thoughts on how we would implement that:
1. in `prepare_table` we add logic to do this
2. also in `prepare_table`
3. we could implement the [clean_up method](https://github.com/meltano/sdk/blob/36f74363f0eaa9c2e2114af6a0d7de2790a20133/singer_sdk/sinks/core.py#L489). 
4. I'm not sure where a try/catch rollback step would fit in right now

@edgarrmondragon any thoughts on where 4 could be done? I'm wondering now if we ultimately want to use this workflow in the future then using a drop/recreate workflow now instead of introducing a new truncate method might be better, since the truncate method would need to be deprecated almost immediately. I considered that at first but didnt know if there were any risks of dropping a table vs truncating it, it felt like truncate was safer. Thoughts?